### ISSUE TITLE
Getting FATAL image scan error for par file

### ISSUE THREAD

**Issue by gloppyuser** [aquasecurity/trivy#3806]
## What did you expect to happen?
Successful scan

## What happened instead?
Scan failed with FATAL error

## Output of run with `-debug`:

```
$ trivy image --offline-scan --format template --template trivy/body.tpl --exit-code 0 --severity "HIGH,CRITICAL" --no-progress --input images/iic-osic-tools_latest.tar --debug
2023-03-09T10:17:03.605-0600	DEBUG	Severities: ["HIGH" "CRITICAL"]
2023-03-09T10:17:03.628-0600	DEBUG	cache dir:  /local/symud/.cache/trivy
2023-03-09T10:17:03.632-0600	DEBUG	DB update was skipped because the local DB is the latest
2023-03-09T10:17:03.632-0600	DEBUG	DB Schema: 2, UpdatedAt: 2023-03-09 12:08:09.70049077 +0000 UTC, NextUpdate: 2023-03-09 18:08:09.70049017 +0000 UTC, DownloadedAt: 2023-03-09 15:32:50.55245154 +0000 UTC
2023-03-09T10:17:03.643-0600	INFO	Vulnerability scanning is enabled
2023-03-09T10:17:03.643-0600	DEBUG	Vulnerability type:  [os library]
2023-03-09T10:17:03.643-0600	INFO	Secret scanning is enabled
2023-03-09T10:17:03.643-0600	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-03-09T10:17:03.643-0600	INFO	Please see also https://aquasecurity.github.io/trivy/v0.38/docs/secret/scanning/#recommendation for faster secret detection
2023-03-09T10:17:43.334-0600	DEBUG	No secret config detected: trivy-secret.yaml
2023-03-09T10:17:43.453-0600	DEBUG	Image ID: sha256:3434c2fc4387e0b447fd0ecbebc11f9dc807f2f4e93194a2e5e45aeda61d6d18
2023-03-09T10:17:43.453-0600	DEBUG	Diff IDs: [sha256:c5ff2d88f67954bdcf1cfdd46fe3d683858d69c2cadd6660812edfc83726c654 sha256:bac19ffcd6c8c9ef527f15da7695ca4891d6531f1b8fadd087687aed75b78457 sha256:1c9ec2e786a95a4538da2261c7b14e4381da312abc63c805c336d45e2947fa15 sha256:26ed29f4f1ae07d0f8a6e33bc769a9a2c258290ee7bfba39ae2be78fc004e089 sha256:47d5b0b1225bea41f08a2943193f67999c2a6fd618b97722e183392c41f7735a sha256:a71394c70065b5ee3c9eda11cf40fdfbbcb4a588e78082c9ba309c3db763cb0f sha256:d65301a81f751be5549a2a14856b420a76d4badb357c7f512777d186e170fc19 sha256:dde2eccd47867075bcebdc134a0b649ce5bd09fa5a84384272659e8ecb9dd99c sha256:f23b78863c6c9c73823cfc335e5e6ff85018e414917b303388af3b9831025165 sha256:9402c8b94c5f79fefe94e44916578fe84b0578429d02e174ef34659045c346ed sha256:204e28fd97b35999925edad9b45c2e1f93f57f665eb9acd1c1498290ce0aa5b7 sha256:4fa59eb5aa565daf117dbdd41659b6e9e4389459b7c9d2c1b3ab695718c88240 sha256:e3ea9f91a233d64e756709a354e86c10cd3a621092e632d978152f876146e61e sha256:a7158990b17fd9af3a40376d66b78e211f13bacab02c6eaf103e79407c045b2f sha256:671326f17a9194c8711b17096105c85feffc7a045559599f56167b4ff585c20c sha256:f2ac95f021cedf130c21de15e8c9183e45c9d776ddd2448e19b3b43e64761683 sha256:75b2056854b56509320d41b6654a29b4acb4aea624091bb6482d0d6f6766d966 sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef sha256:31b9e7e35ecafb86653c92ada149bf8d24dff4cc7d4c3b54fa23473fb152d2a9 sha256:2e26eee7a44c60aecf7fd4722c9aaeea689e9854bac33f560dc117be891748a7 sha256:cfc2161889186c616dd88ab4805f36f173c453b7748d0b80731fdd52cfad9125 sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef sha256:6b6971aa0b770927a2ea295141dd4abfcfb59913774f0df9f0eeecd9ad14358b sha256:c8cf8af419105b79b9ba2c6c5c83404b65722992254d1957e54f682a74bf78d8 sha256:1d1e421b63153ce9437d669a5436a41af28268cae69f3ddbdcc08aa291eb7ddb sha256:940da3486d9c6e31ec01019a33249f8a0f8550ccef9482791b9dddce856f308f sha256:e3a85705e6502a599df980a63d04e81f67233aa40de3585915812480d8fce835 sha256:82cb47fc08295f3f8353d3873d680b606952201b58a99a96c4f271e2fec55ebc sha256:ddedd9b17881a2da2b3a44674d5d7423ae22a6321deacfd0bb3adeb85c4d59a0 sha256:c02103a99c7097600df1cfefde2eb6db5db0ba183dcd91ddd0b343c812f4a840 sha256:9543975f297bd45ceb18cb73d6763616f5417e7acfd0948a806e8bdcd4f0e597 sha256:a07b23beade02b765aa05d1ed2e2816169296f266ba6577c6a95521e3784c963 sha256:f8921a19529650e6d227759a24aeda6cb4290e02add5056e5e5d7f952aefe9ff sha256:efb6befaab8bb582672bfd59f75eb0d3004c8b7c5ee112df2de2d63dbfe28fd2 sha256:eabe29d1b13590da40881595a4662dab56630c610867d0c23d24b1c32a31463f sha256:a8c4a0086e955b3641185f16297c11e85f33f12fdb47d14a8779186c523ee697 sha256:b423b839fca209a1f4add6d07d6712c26e06270ef3b0aeb12f490df7087171c7 sha256:844b72371899b79d194a9ce3c8c2d87c0816146e094198a8b9730aa4dad8ab5c sha256:b662ca863aa81d120c5a2661c9e5e0ef7cebb33f388ce1493782fdfee9ec2f2f sha256:9555b5f9152caaa281066bfa4b2685ca29f42c49664c676d04dbef7e218af0bc sha256:c86fbefdecce22f5ccd10d97db9ea53d956809b1c924a96eb611a219eebc14af sha256:673bf0a69c2e770e0c40a488954fca26eb0c79d5d6cfa73e3b61899e4eeafca7 sha256:dcc60c8ffe121fe3e1cd5eb69de85c8ce05820f504561c829c82473e2cc2b32c sha256:97ee4b693232c2abcf3356effcdbc27f0fcb2207de081e07149586220a0acc3d sha256:e86001f53a0f2a4dc43921fd220a15a6d939cbfa758f43e466b861eb30d157ad sha256:be048d97acddc48f5e0b1b22571f7bd55543a9c5fe1ca25c9d38b0e5f8004975 sha256:e9db874311570448b063a5fcfda280ade29c64b0f7c27d6225bb346769c47779 sha256:6156b3dec1e381281089744ca2f2b2210457f2878bcfcd88e4835197d654e03f sha256:3332099634bcde7dbee27704ed532e02266634634be2ee2e96efa3a586019cee sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef]
2023-03-09T10:17:43.454-0600	DEBUG	Base Layers: [sha256:c5ff2d88f67954bdcf1cfdd46fe3d683858d69c2cadd6660812edfc83726c654]
2023-03-09T10:17:44.918-0600	DEBUG	Missing image ID in cache: sha256:3434c2fc4387e0b447fd0ecbebc11f9dc807f2f4e93194a2e5e45aeda61d6d18
2023-03-09T10:17:44.918-0600	DEBUG	Missing diff ID in cache: sha256:dde2eccd47867075bcebdc134a0b649ce5bd09fa5a84384272659e8ecb9dd99c
2023-03-09T10:19:41.422-0600	INFO	JAR files found
2023-03-09T10:19:41.565-0600	INFO	Analyzing JAR files takes a while...
2023-03-09T10:19:41.571-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_hd.par"}
2023-03-09T10:19:41.571-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_lp.par"}
2023-03-09T10:19:41.580-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_ms.par"}
2023-03-09T10:19:41.571-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_hdll.par"}
2023-03-09T10:19:41.570-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_hs.par"}
2023-03-09T10:19:41.570-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_osu_sc_12t_hs.par"}
2023-03-09T10:19:41.572-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_ls.par"}
2023-03-09T10:19:41.570-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/gf180mcu/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/gf180mcuC/libs.tech/qflow/gf180mcu_fd_sc_mcu7t5v0.par"}
2023-03-09T10:19:41.571-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/gf180mcu/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/gf180mcuC/libs.tech/qflow/gf180mcu_fd_sc_mcu9t5v0.par"}
2023-03-09T10:19:41.582-0600	DEBUG	Parsing Java artifacts...	{"file": "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_hvl.par"}
2023-03-09T10:19:41.584-0600	FATAL	image scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:427
  - scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanArtifact
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:266
  - scan failed:
    github.com/aquasecurity/trivy/pkg/commands/artifact.scan
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:669
  - failed analysis:
    github.com/aquasecurity/trivy/pkg/scanner.Scanner.ScanArtifact
        /home/runner/work/trivy/trivy/pkg/scanner/scan.go:146
  - analyze error:
    github.com/aquasecurity/trivy/pkg/fanal/artifact/image.Artifact.Inspect
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/image/image.go:139
  - failed to analyze layer (sha256:dde2eccd47867075bcebdc134a0b649ce5bd09fa5a84384272659e8ecb9dd99c):
    github.com/aquasecurity/trivy/pkg/fanal/artifact/image.Artifact.inspect.func1
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/image/image.go:242
  - post analysis error:
    github.com/aquasecurity/trivy/pkg/fanal/artifact/image.Artifact.inspectLayer
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/image/image.go:320
  - post analysis error:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer.AnalyzerGroup.PostAnalyze
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/analyzer.go:486
  - walk dir error:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer/language/java/jar.(*javaLibraryAnalyzer).PostAnalyze
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/language/java/jar/jar.go:91
  - on file:
    github.com/aquasecurity/trivy/pkg/parallel.walk[...]
        /home/runner/work/trivy/trivy/pkg/parallel/walk.go:97
  - "foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow/sky130_fd_sc_ms.par" parse error:
    github.com/aquasecurity/trivy/pkg/fanal/analyzer/language/java/jar.(*javaLibraryAnalyzer).PostAnalyze.func2
        /home/runner/work/trivy/trivy/pkg/fanal/analyzer/language/java/jar/jar.go:75
  - zip error:
    github.com/aquasecurity/go-dep-parser/pkg/java/jar.(*Parser).parseArtifact
        /home/runner/go/pkg/mod/github.com/aquasecurity/go-dep-parser@v0.0.0-20230302111817-e4068021315b/pkg/java/jar/parse.go:82
  - zip: not a valid zip file
```

## Output of `trivy -v`:

```
Version: 0.38.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-03-09 12:08:09.70049077 +0000 UTC
  NextUpdate: 2023-03-09 18:08:09.70049017 +0000 UTC
  DownloadedAt: 2023-03-09 15:32:50.55245154 +0000 UTC

```

## Additional details (base image name, container registry info...):
hpretl/iic-osic-tools:latest from dockerhub registry.

**Comment by DmitriyLewen** [comment]
Hello @gloppyuser 
thanks for your report!

Looks like your image contains wrong jar files:
```
➜  pwd
/foss/pdks/volare/sky130/versions/032b059033c4cf67f94c5b7f0a44d936b8ff5aae/sky130A/libs.tech/qflow
➜  jar -xf sky130_fd_sc_hd.par 
java.util.zip.ZipException: zip END header not found
	at java.base/java.util.zip.ZipFile$Source.zerror(ZipFile.java:1470)
	at java.base/java.util.zip.ZipFile$Source.findEND(ZipFile.java:1371)
	at java.base/java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1378)
	at java.base/java.util.zip.ZipFile$Source.<init>(ZipFile.java:1209)
	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1172)
	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:719)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:239)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:169)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:140)
	at jdk.jartool/sun.tools.jar.Main.extract(Main.java:1388)
	at jdk.jartool/sun.tools.jar.Main.run(Main.java:410)
	at jdk.jartool/sun.tools.jar.Main.main(Main.java:1680)
➜  java -jar sky130_fd_sc_hd.par
Error: Invalid or corrupt jarfile sky130_fd_sc_hd.par
```

That is why Trivy returns error for this image.

**Comment by gms1** [comment]
it cannot be assumed that every par file is also a zip file, there are other reasons a file might have the par extension

**Comment by DmitriyLewen** [comment]
Hello @gms1 

There is #4037 to stop returning errors for this case.

**Comment by knqyf263** [comment]
#4037 was shipped in v0.41.0
